src/Currency/index.ts

```diff
@@ -3,46 +3,86 @@
  */
 import * as E from "fp-ts/Either";
 import { Lazy, flow } from "fp-ts/function";
-import { lazy, isNumber, isNotNil } from "../internal";
+import { lazy, isNumber, isNotNil, isString } from "../internal";
 import { INumberFormat } from "../Number";
+import { id, ILocale } from "../locale";
 
 interface ICurrencyFormat extends INumberFormat {
   style: Lazy<"currency">;
 }
 
-const currencyFormatter = () =>
-  flow(
+const currencyFormatter = (locale: ILocale, precision = 0) => {
+  const { currency, delimiters } = locale;
+  const { symbol } = currency;
+  const { thousands, decimal } = delimiters;
+
+  const toStringFormat = flow(
+    (value: number) =>
+      `${symbol} ` +
+      value
+        .toFixed(precision)
+        .replace(/\./g, decimal)
+        .replace(/(\d)(?=(\d{3})+(?!\d))/g, `$1${thousands}`)
+  );
+
+  return flow(
```

Comment:
is it posisble to do this?

we define the `E.left` handler upfront (also on line55)
```
const onlyNotNil = flow(
   isNotNil,
   E.chain(a => E.right(a)),
   E.mapLeft(() => new TypeError("invalid format"))
)
```
then the composing function is free from handling the `E.left`
```
return flow(
    onlyNotNil,
    E.chain(isNumber),
    E.chain(toStringFormat)
)
```

Reply:
this is possible but I think it's not preferable, onlyNotNill will only applicable to currency formatter since it throwing error, so it can't be reused.

and the message `invalid format` here no only applied to isNotNil, but also applied to `isNumber`

Reply:
yes, I mean the `onlyNotNill` is just like a helper of the `/Currency` and will be isolated within this context. Therefore, the helper can be used to compose the `currencyFormater` and `currenceyParse`, so we don't need to handle the `E.left` on each function

Reply:
still om, we need to handle left because of `isNumber`

Reply:
ok, make sense